cronet/src/main/java/io/grpc/cronet/CronetCallOptions.java

```diff
@@ -38,25 +38,26 @@ private CronetCallOptions() {}
       CallOptions.Key.of("cronet-annotation", null);
 
   /**
-   * Returns a copy of {@code callOptions} with {@code annotation} included in the set of Cronet
+   * Returns a copy of {@code callOptions} with {@code annotation} included as one of the Cronet
    * annotation objects. When an RPC is made using a {@link CallOptions} instance returned by this
    * method, the annotation objects will be attached to the underlying Cronet bidirectional stream.
-   * When the stream finishes, the user can retrieve the annotation object(s) via {@link
+   * When the stream finishes, the user can retrieve the annotation objects via {@link
    * org.chromium.net.RequestFinishedInfo.Listener}.
    *
    * @param annotation the object to attach to the Cronet stream
    */
   public static CallOptions withAnnotation(CallOptions callOptions, Object annotation) {
-    Set<Object> annotations = callOptions.getOption(CRONET_ANNOTATIONS_KEY);
-    if (annotations != null) {
-      annotations.add(annotation);
+    Collection<Object> existingAnnotations = callOptions.getOption(CRONET_ANNOTATIONS_KEY);
+    ArrayList<Object> newAnnotations;
+    if (existingAnnotations == null) {
+      newAnnotations = new ArrayList<Object>();
     } else {
-      annotations = new HashSet<Object>();
-      annotations.add(annotation);
+      newAnnotations = new ArrayList<Object>(existingAnnotations);
     }
-    return callOptions.withOption(CronetCallOptions.CRONET_ANNOTATIONS_KEY, annotations);
+    newAnnotations.add(annotation);
+    return callOptions.withOption(CronetCallOptions.CRONET_ANNOTATIONS_KEY, newAnnotations);
```

Comment:
You may consider wrapping with unmodifiableList.

Reply:
Done